src/node_crypto.cc

```diff
@@ -3952,6 +3954,37 @@ void SignBase::CheckThrow(SignBase::Error error) {
   }
 }
 
+int SignBase::GetRSAOptions(Environment* env, v8::Local<v8::Object> options,
```

Comment:
> I guess I should move default values to JS as well, such that the native code does not need to handle undefined values.

Yes, that's a good idea.

> it might be necessary to add more options in the future and all of these are optional, meaning that we might end up passing useless arguments most of the time.

Efficiency-wise a few extra parameters won't break the bank (they're just stack slots) and like Anna says, the cost of crypto itself probably dwarfs the overhead.